### PR TITLE
vectorstore: Add support to filter by IS NULL and IS NOT NULL criteria

### PR DESCRIPTION
**Pull request by MartinGotelli**
Description: Add support to filter by IS NULL criteria in the metadata
Issue: N/A
Dependencies: N/A
Twitter handle: @martinferenaz

This PR fixes the problem when you want to search if a tag has a value (exists) in the metadata. To check that you need to compare with the "IS NULL" method and the "$eq" operator only checks that the tag exists and has a null value assigned.

In addition, a test for the $nin condition was added and a bug arrised, the bug was resolved adding a not condition after the in_

### CODE REVIEW

File: langchain_postgres/vectorstores.py [hunk 1]
```diff
@@ -702,13 +703,24 @@ def _handle_field_filter(
...
             elif operator in {"$nin"}:
-                return queried_field.nin_([str(val) for val in filter_value])
+                return ~queried_field.in_([str(val) for val in filter_value])
             elif operator in {"$like"}:
                 return queried_field.like(filter_value)
             elif operator in {"$ilike"}:
                 return queried_field.ilike(filter_value)
             else:
                 raise NotImplementedError()
+        elif operator == "$null":
```
Comment (MartinGotelli) [inline langchain_postgres/vectorstores.py:713]: @eyurtsev let me know if you think that this operator should be called something like "$exists"

File: langchain_postgres/vectorstores.py [hunk 2]
```diff
@@ -702,13 +703,24 @@ def _handle_field_filter(
             if operator in {"$in"}:
                 return queried_field.in_([str(val) for val in filter_value])
             elif operator in {"$nin"}:
-                return queried_field.nin_([str(val) for val in filter_value])
+                return ~queried_field.in_([str(val) for val in filter_value])
```
Comment (eyurtsev) [inline langchain_postgres/vectorstores.py:706]: thank you!

File: langchain_postgres/vectorstores.py (hunk 1 above)
Comment (eyurtsev) [reply langchain_postgres/vectorstores.py:713]: Yes, would you mind updating to $exists.

**Review by eyurtsev — COMMENTED** [review]
Looks great! I can merge and release as soon as $null is updated to $exists

**Review by eyurtsev — APPROVED** [review]

**Pull request merged**